pkgs/unified_analytics/lib/src/event.dart

```diff
@@ -716,11 +719,93 @@ final class Event {
       other is Event &&
       other.runtimeType == runtimeType &&
       other.eventName == eventName &&
-      compareEventData(other.eventData, eventData);
-
-  @override
-  String toString() => jsonEncode({
+      _compareEventData(other.eventData, eventData);
+
+  /// Converts an instance of [Event] to JSON.
+  ///
+  /// Example for [Event.timing] converted to JSON below.
+  /// ```json
+  /// {
+  ///   "eventName": "timing",
+  ///   "eventData": {
+  ///     "workflow": "my-work-flow",
+  ///     "variableName": "my-variable",
+  ///     "elapsedMilliseconds": 123,
+  ///     "label": "my-label"
+  ///   }
+  /// }
+  /// ```
+  String toJson() => jsonEncode({
         'eventName': eventName.label,
         'eventData': eventData,
       });
+
+  @override
+  String toString() => toJson();
+
+  /// Utility function to take in two maps [a] and [b] and compares them
+  /// to ensure that they have the same keys and values.
+  ///
+  /// Used for the equality operator.
+  bool _compareEventData(Map<String, Object?> a, Map<String, Object?> b) {
```

Comment:
instead of creating our own, can we use something from package:collection? like `DeepCollectionEquality`?

Reply:
Good idea, [this commit](https://github.com/dart-lang/tools/pull/251/commits/2d58a3db445ecc36bfe148ac22cd49e34e182ef4) has it updated to use `DeepCollectionEquality.equals` to compare instead

@christopherfujino, i know we discussed trimming down dependencies for this package, do you think this is a valid reason to add another dep?

Reply:
Following up on offline discussions with @christopherfujino, going to remove the dep on collection and use the same function for comparing two separate maps to reduce dependencies for this package